torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -521,10 +553,12 @@ void TensorPipeAgent::sendCompletedResponseMessage(
       }
     }
 
+    // TODO: pass streams to pipeWrite
```

Comment:
From the way I see that `sendCompletedResponseMessage` is called, it looks like the `streams[i]` is the stream used to read the `i`-th input tensor. I don't see how we can, from that, figure out which stream we should send the `i`-th _output_ tensor on. Even if we could query which device a stream is for (which I'm not sure is possible), we could be in a situation where there were multiple input streams for a device, or none.

I think a reasonable solution could be to have one stream per device, not one per input tensor.

Reply:
yep, the plan is to have one stream per device.